Update documentation with changes related to metrics-server fix (#7473)
prometheus-operator/prometheus-operator: docs/remove-metrics-server.md

# ⚠️ Remove `metrics-server` Before Using `kube-prometheus`

> Before setting up `kube-prometheus`, ensure that the `metrics-server` is **removed** from your Kubernetes cluster to avoid conflicts.

Reviewer: ```suggestion
Before installing `kube-prometheus`, in order to prevent conflicts you should ensure that the `metrics-server` is **removed** from your Kubernetes cluster.
```

## Why Remove `metrics-server`?

The `metrics-server` can interfere with the Prometheus Adapter's custom metrics endpoints (`/apis/custom.metrics.k8s.io`). This may lead to issues with metric scraping and cause Prometheus Adapter errors. 

> **Steps to Remove `metrics-server`:**

Reviewer: Considering the manifest defined in the instalation https://github.com/kubernetes-sigs/metrics-server?tab=readme-ov-file#installation to fully remove metrics server there's additional steps. You can check more details https://github.com/prometheus-operator/kube-prometheus/issues/926#issuecomment-1228132910
